sql/src/main/java/io/crate/execution/engine/collect/sources/InformationSchemaIterables.java

```diff
@@ -126,28 +125,46 @@ public InformationSchemaIterables(final Schemas schemas,
         clusterService.addListener(this);
     }
 
+    private static Stream<ViewInfo> viewsStream(Schemas schemas) {
+        return sequentialStream(schemas)
+            .flatMap(schema -> sequentialStream(schema.getViews()))
+            .filter(i -> !IndexParts.isPartitioned(i.ident().indexName()));
+    }
+
+    private static Stream<TableInfo> tablesStream(Schemas schemas) {
+        return sequentialStream(schemas)
+            .flatMap(s -> sequentialStream(s.getTables()))
+            .filter(i -> !IndexParts.isPartitioned(i.ident().indexName()));
+    }
+
+    private static <T> Stream<T> sequentialStream(Iterable<T> iterable) {
+        return stream(iterable.spliterator(), false);
+    }
+
     public Iterable<SchemaInfo> schemas() {
         return schemas;
     }
 
     public Iterable<RelationInfo> relations() {
-        return FluentIterable.concat(tables, views);
+        return relations.get().collect(toList());
```

Comment:
we could avoid the collect calls everywhere by using a method reference to iterator to create an iterable. That way the whole transformation chain keeps being lazy

Reply:
hm, that's a good idea. I'm assuming you're thinking of something along the lines of `() -> relations.get().iterator();` ? Or do you have a different thing in mind? (the method reference passing is what I don't really get)

Reply:
Ah I see what you mean

Reply:
method reference doesn't always work as subsequent `Iterable` iterators should operate on new streams, but this was a great idea as I managed to avoid the stream suppliers. thanks